### PR TITLE
Mount custom eec tls certificate

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

- This PR implements Mount custom EEC TLS certificate to eec sts and otel sts.

## How can this be tested?

1. Create a secret with custom TLS for eec

```
openssl req -nodes -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -subj "/CN=dynakube-extensions-controller.dynatrace"
kubectl create secret tls custom-tls-cert --cert=cert.pem --key=key.pem
```

2. Deploy dynakube:

```
spec:
  extensions:
    prometheus:
      enabled: true
  templates:
    extensionExecutionController:
      tlsRefName: custom-tls-cert
      imageRef:
        repository: extk8sregistry.azurecr.io/eec/dynatrace-eec
        tag: 1.300.0.20240814-073129
```

3. Make sure all secret are mounted properly and env vars are there based on 
ticket description [K8S-10666](https://dt-rnd.atlassian.net/browse/K8S-10666)

### CODE REVIEW

File: pkg/controllers/dynakube/extension/eec/statefulset.go [hunk 1]
```diff
@@ -24,27 +24,34 @@ const (
...
 	envK8sClusterId                 = "K8sClusterUID"
 	envActiveGateTrustedCertName    = "ActiveGateTrustedCert"
-	envActiveGateTrustedCert        = "/var/lib/dynatrace/secrets/ag/server.crt"
 	envK8sExtServiceUrl             = "K8sExtServiceUrl"
 	envHttpsCertPathPem             = "HttpsCertPathPem"
 	envHttpsPrivKeyPathPem          = "HttpsPrivKeyPathPem"
 	envDSTokenPath                  = "DSTokenPath"
-
-	tokensVolumeName                   = "tokens"
+	envEECHttpsCertPathPem          = "HttpsCertPathPem"
```
Comment (aorcholski) [inline pkg/controllers/dynakube/extension/eec/statefulset.go:42]: it's duplicate of 39-40
```
envHttpsCertPathPem             = "HttpsCertPathPem"
envHttpsPrivKeyPathPem          = "HttpsPrivKeyPathPem"
```
And related Volumes and volumeMouts are duplicated.

File: pkg/controllers/dynakube/extension/eec/statefulset.go [hunk 2]
```diff
@@ -353,6 +379,27 @@ func setVolumes(dk *dynakube.DynaKube) func(o *appsv1.StatefulSet) {
 			})
 		}
 
+		if dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+			o.Spec.Template.Spec.Volumes = append(o.Spec.Template.Spec.Volumes, corev1.Volume{
+				Name: eecTlsCertificatesVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: dk.Spec.Templates.ExtensionExecutionController.TlsRefName,
```
Comment (aorcholski) [inline pkg/controllers/dynakube/extension/eec/statefulset.go:387]: just use this name in a condition in line 334

This Volume duplicates 330-348

**Review by aorcholski — CHANGES_REQUESTED** [review]

File: pkg/controllers/dynakube/extension/eec/statefulset.go (hunk 2 above)
Comment (andriisoldatenko) [reply pkg/controllers/dynakube/extension/eec/statefulset.go:387]: on it.

File: pkg/controllers/dynakube/extension/eec/statefulset.go [hunk 3]
```diff
@@ -273,11 +272,33 @@ func buildContainerVolumeMounts(dk *dynakube.DynaKube) []corev1.VolumeMount {
...
 }
 
 func setVolumes(dk *dynakube.DynaKube) func(o *appsv1.StatefulSet) {
 	return func(o *appsv1.StatefulSet) {
+		tlsSecretName := extensionsControllerTlsSecretName
+		tlsCertVolumeName := httpsCertVolumeName
+
+		if dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+			tlsSecretName = dk.Spec.Templates.ExtensionExecutionController.TlsRefName
+			tlsCertVolumeName = consts.ExtensionsCustomTlsCertificate
```
Comment (aorcholski) [inline pkg/controllers/dynakube/extension/eec/statefulset.go:299]: why two different Volume names are used?

File: pkg/controllers/dynakube/extension/eec/statefulset.go [hunk 4]
```diff
@@ -273,11 +272,33 @@ func buildContainerVolumeMounts(dk *dynakube.DynaKube) []corev1.VolumeMount {
 		})
 	}
 
+	if dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      consts.ExtensionsCustomTlsCertificate,
+			MountPath: baseSecretsPath + "/eec",
```
Comment (aorcholski) [inline pkg/controllers/dynakube/extension/eec/statefulset.go:278]: Why two different mountpoints are used? Eventually it's secret created by operator or user. Just different name of the secret.

File: pkg/controllers/dynakube/extension/eec/statefulset.go [hunk 5]
```diff
@@ -56,11 +59,14 @@ const (
 	activeGateTrustedCertMountPath     = "/var/lib/dynatrace/secrets/ag"
 	activeGateTrustedCertSecretKeyPath = "server.crt"
 	httpsCertVolumeName                = "https-certs"
-	httpsCertFile                      = "cert.pem"
-	httpsPrivKeyFile                   = "key.pem"
 	httpsCertMountPath                 = "/var/lib/dynatrace/remotepluginmodule/secrets/https/"
```
Comment (StefanHauth) [inline pkg/controllers/dynakube/extension/eec/statefulset.go:62]: The last `/` leads to double dashes in the full file path later on, plz. remove it while you're at it.

File: pkg/controllers/dynakube/extension/eec/statefulset.go (hunk 4 above)
Comment (StefanHauth) [reply pkg/controllers/dynakube/extension/eec/statefulset.go:278]: I agree. The mount point should always be the same, whether the cert is created by the user or by the operator.

File: pkg/controllers/dynakube/extension/eec/statefulset.go (hunk 3 above)
Comment (StefanHauth) [reply pkg/controllers/dynakube/extension/eec/statefulset.go:299]: Same as above, it shouldn't make a difference who creates the cert.

**Review by StefanHauth — CHANGES_REQUESTED** [review]

File: pkg/controllers/dynakube/extension/eec/statefulset.go [hunk 6]
```diff
@@ -210,15 +209,20 @@ func buildContainerEnvs(dk *dynakube.DynaKube) []corev1.EnvVar {
...
+		{Name: envHttpsPrivKeyPathPem, Value: httpsCertMountPath + "/" + consts.TLSKeyDataName},
 		{Name: envDSTokenPath, Value: dsTokenPath},
 	}
 
 	if dk.Spec.ActiveGate.TlsSecretName != "" {
 		containerEnvs = append(containerEnvs, corev1.EnvVar{Name: envActiveGateTrustedCertName, Value: envActiveGateTrustedCert})
 	}
 
+	if dk.Spec.Templates.ExtensionExecutionController.TlsRefName != "" {
+		containerEnvs = append(containerEnvs, corev1.EnvVar{Name: envHttpsCertPathPem, Value: envEecHttpsCertPathPem})
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply pkg/controllers/dynakube/extension/eec/statefulset.go:220]: fixed in 36b9bc48

File: pkg/controllers/dynakube/extension/eec/statefulset.go (hunk 3 above)
Comment (andriisoldatenko) [reply pkg/controllers/dynakube/extension/eec/statefulset.go]: fixed in bc76d466

File: pkg/controllers/dynakube/extension/eec/statefulset.go (hunk 4 above)
Comment (andriisoldatenko) [reply pkg/controllers/dynakube/extension/eec/statefulset.go:279]: fixed in bc76d466

File: pkg/controllers/dynakube/extension/eec/statefulset.go (hunk 5 above)
Comment (andriisoldatenko) [reply pkg/controllers/dynakube/extension/eec/statefulset.go]: fixed in bc76d466

File: pkg/controllers/dynakube/extension/eec/reconciler_test.go [hunk 7]
```diff
@@ -672,7 +666,26 @@ func TestUpdateStrategy(t *testing.T) {
...
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "custom-tls",
+				},
+			},
+		}
+
+		require.Contains(t, statefulSet.Spec.Template.Spec.Volumes, expectedVolume)
+	})
+}
```
Comment (aorcholski) [inline pkg/controllers/dynakube/extension/eec/reconciler_test.go:686]: could you please move Volume tests from 304-464 to new TestVolumes function ?

File: pkg/controllers/dynakube/extension/eec/statefulset.go [hunk 8]
```diff
@@ -24,27 +24,29 @@ const (
 	containerName                    = "extensions-controller"
 	collectorPort                    = int32(14599)
 	serviceAccountName               = "dynatrace-extensions-controller"
+	baseSecretsPath                  = "/var/lib/dynatrace/secrets"
```
Comment (aorcholski) [inline pkg/controllers/dynakube/extension/eec/statefulset.go:27]: unused, please remove it

**Review by aorcholski — CHANGES_REQUESTED** [review]

File: pkg/controllers/dynakube/extension/eec/reconciler_test.go (hunk 7 above)
Comment (andriisoldatenko) [reply pkg/controllers/dynakube/extension/eec/reconciler_test.go:685]: fixed in 6280db93

**Review by aorcholski — APPROVED** [review]
LGTM

**Review by StefanHauth — APPROVED** [review]
LGTM based on what I can see in the deployed EEC and OTelC manifests.

**Pull request merged**